### PR TITLE
Missing ItemType Functions

### PR DESCRIPTION
**Pull request by Codinablack**
The itemtype functions I listed here

https://github.com/otland/forgottenserver/issues/2293

### CODE REVIEW

File: src/luascript.cpp [hunk 1]
```diff
@@ -10667,6 +10676,19 @@ int LuaScriptInterface::luaItemTypeCreate(lua_State* L)
 	return 1;
 }
 
+int LuaScriptInterface::luaItemTypeIsBlocking(lua_State* L)
+{
+	// itemType:isBlocking()
+	const ItemType* itemType = getUserdata<const ItemType>(L, 1);
+	if (itemType) {
+		pushBoolean(L, itemType->blockProjectile || itemType->blockSolid);
+	}
```
Comment (wgrriffel) [inline src/luascript.cpp]: tfs indents else java like, no new line after closing bracket

Comment (ranisalt) [inline src/luascript.cpp]: ugh, don't you compare it with Java, it's offensive

Comment (Codinablack) [inline src/luascript.cpp]: I'm not seeing the difference between this and every other instance this occurs in functions throughout this file... what did I do wrong here? 10685 has a bracket 10686 has else {

Comment (wgrriffel) [inline src/luascript.cpp]: @ranisalt unfortunately its not a comparison, is how this indentation style is named :rofl:

Comment (ranisalt) [inline src/luascript.cpp]: @Codinablack `} else {`, in a single line.

Comment (Codinablack) [inline src/luascript.cpp]: I need to do that to all the functions? If so then it needs to be done to all of them in that file, not just the ones I added

Comment (wgrriffel) [inline src/luascript.cpp]: no, functions are indented with new line brackets

**Comment by Kamenuvol** [comment]
Order the functions alphabetically, please.

**Comment by Codinablack** [comment]
Can I not close a pull request? I deleted my repository because I made a mistake I didn't know how to fix, and now I am working on remaking this so all the functions work, and the other flaws are fixed.

**Comment by Codinablack** [comment]
@Kamenuvol The methods aren't organized alphabetically as is right now on the master branch here. I just tried to insert where I thought the methods seemed most appropriate, relative to the methods type and purpose. Should I reorganize them all in the next PR?

**Comment by ranisalt** [comment]
Keep them grouped by scope.

**Comment by Codinablack** [comment]
@ranisalt I'm not sure what you mean. I'm not much of a c++ guy, but I understand scope. Far as I can tell the methods are all in the same scope because they are all for the same class type. 

Traditionally when adding people have added to the bottom, for the is methods example like this

`	registerMethod("ItemType", "isCorpse", LuaScriptInterface::luaItemTypeIsCorpse);
	registerMethod("ItemType", "isDoor", LuaScriptInterface::luaItemTypeIsDoor);
	registerMethod("ItemType", "isContainer", LuaScriptInterface::luaItemTypeIsContainer);
	registerMethod("ItemType", "isFluidContainer", LuaScriptInterface::luaItemTypeIsFluidContainer);
	registerMethod("ItemType", "isMovable", LuaScriptInterface::luaItemTypeIsMovable);
	registerMethod("ItemType", "isRune", LuaScriptInterface::luaItemTypeIsRune);
	registerMethod("ItemType", "isStackable", LuaScriptInterface::luaItemTypeIsStackable);
	registerMethod("ItemType", "isReadable", LuaScriptInterface::luaItemTypeIsReadable);
	registerMethod("ItemType", "isWritable", LuaScriptInterface::luaItemTypeIsWritable);
	registerMethod("ItemType", "isBlocking", LuaScriptInterface::luaItemTypeIsBlocking);
	registerMethod("ItemType", "isGroundTile", LuaScriptInterface::luaItemTypeIsGroundTile);
	registerMethod("ItemType", "isMagicField", LuaScriptInterface::luaItemTypeIsMagicField);
	registerMethod("ItemType", "isUseable", LuaScriptInterface::luaItemTypeIsUseable);
	registerMethod("ItemType", "isRotateable", LuaScriptInterface::luaItemTypeIsRotateable);
	registerMethod("ItemType", "isPickupable", LuaScriptInterface::luaItemTypeIsPickupable);`

Does this approach fit the guidelines or requested standards? With the get type added to the bottom of that list, and has beneath the only other has method for itemtype..?

**Comment by Codinablack** [comment]
I believe I have the commit ready, before I create a new PR (sorry that I deleted old repository) could I get some feedback on the commit.

https://github.com/Codinablack/forgottenserver/commit/006390a28afc0657925fe159f90d705e4823a00d

**Pull request closed without merging**